### PR TITLE
Initial commit of spot-electron

### PR DESCRIPTION
**Pull request by zbettenbuk**
Initial commit of the electron app with the loadURL approach as we agreed.

This now supports
- Persistent config
- Window mode changes
- Logging

### CODE REVIEW

File: spot-electron/src/logger/logger.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+     * @param {?Error} throwable - The error to be logged, if any.
+     * @returns {void}
+     */
+    _log(level, message, throwable) {
+        // eslint-disable-next-line no-console
+        console.log(`${new Date().toISOString()} - ${level} - ${message}${throwable ? ` - ${throwable}` : ''}`);
+    }
+}
+
+exports.logger = new Logger();
```
Comment (virtuacoplenny) [inline spot-electron/src/logger/logger.js]: Here are some random musings I have about logging. In the future should we be sending these logs somewhere like spot-client does? Should we store them locally? I think in the future we should also add an uncaught exception handler (I forget how I did that previously). Also, should we be using jitsi-meet-logger here too?

File: spot-electron/Makefile [hunk 2]
```diff
@@ -0,0 +1,3 @@
+default:
+	# Build spot-client and get the artifacts here
+	cd ../spot-client && npm run build -- --env.target=../spot-electron
```
Comment (virtuacoplenny) [inline spot-electron/Makefile]: Do you still need this makefile?

File: spot-electron/config.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+module.exports = {
+
+    /**
+     * The default URL to connect to.
+     */
+    defaultSpotURL: 'https://spot.8x8.vc/spot',
```
Comment (virtuacoplenny) [inline spot-electron/config.js]: Maybe add a process.env variable right now to make this configurable immediately? That would be my preference, but not a requirement.

File: spot-electron/package.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
+{
+    "author": "Jitsi",
+    "build": {
+        "appId": "org.jitsi.spot",
+        "mac": {
+            "category": "public.app-category.business",
+            "target": "default"
+        },
+        "productName": "Jitsi Spot"
```
Comment (virtuacoplenny) [inline spot-electron/package.json]: Is it Jitsi Spot or Jitsi-Meet Spot? This repo is called jitsi-meet-spot. We should get verification at some point in the future.

File: spot-electron/package.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        "eslint": "5.16.0",
+        "eslint-config-jitsi": "github:jitsi/eslint-config-jitsi#1.0.1",
+        "eslint-plugin-flowtype": "3.8.2",
+        "eslint-plugin-import": "2.17.2",
+        "eslint-plugin-jsdoc": "4.8.4"
+    },
+    "license": "ISC",
+    "main": "main.js",
+    "name": "spot-electron",
+    "productName": "Jitsi Spot",
```
Comment (virtuacoplenny) [inline spot-electron/package.json]: Do you need productName here and under build?

File: spot-electron/src/application-window/applicationwindow.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+const {
+    defaultSpotURL,
+    spashScreen
+} = require('../../config');
+
+const { config } = require('../config');
+
+/**
+ * This is the reference for the main window. May be exposed from this scope later, but we don't have
+ * any reasons to di it yet.
```
Comment (virtuacoplenny) [inline spot-electron/src/application-window/applicationwindow.js]: to *hide it yet?

File: spot-electron/Makefile (hunk 2 above)
Comment (zbettenbuk) [reply spot-electron/Makefile]: Nope, thanks!

File: spot-electron/config.js (hunk 3 above)
Comment (zbettenbuk) [reply spot-electron/config.js]: Good call, thanks!

File: spot-electron/package.json (hunk 4 above)
Comment (zbettenbuk) [reply spot-electron/package.json]: Asked Emil, hopefully we get an answer in a few weeks

File: spot-electron/src/application-window/applicationwindow.js (hunk 6 above)
Comment (zbettenbuk) [reply spot-electron/src/application-window/applicationwindow.js]: to *do it yet

thanks!

File: spot-electron/package.json (hunk 5 above)
Comment (zbettenbuk) [reply spot-electron/package.json]: I thought we do, one for the app and one for the packager, but I removed the one under build and it seems like picking up the one on the top level. Thanks!

File: spot-electron/.eslintrc [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+        "eslint-config-jitsi/jsdoc"
+    ],
+    "rules": {
+        "no-console": 1,
+        "space-before-function-paren": ["error", {
+            "anonymous": "always",
+            "named": "never",
+            "asyncArrow": "always"
+        }]
+    },
```
Comment (virtuacoplenny) [inline spot-electron/.eslintrc]: Why the rule override for space-before-function-paren? I wonder why I did the same for spot-webdriver...

File: spot-electron/config.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+     * The default URL to connect to.
+     */
+    defaultSpotURL: process.env.SPOT_URL || 'https://spot.8x8.vc/spot',
+
+    /**
+     * The details of the spash screen, if any.
+     */
+    spashScreen: {
+        height: 192,
+        logo: 'static/applogo.png',
```
Comment (virtuacoplenny) [inline spot-electron/config.js]: Is this the jitsi logo?

File: spot-electron/package.json [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    "scripts": {
+        "dist": "electron-builder",
+        "lint": "eslint .",
+        "pack": "electron-builder --dir",
+        "start": "electron ."
+    },
+    "version": "1.0.0",
```
Comment (virtuacoplenny) [inline spot-electron/package.json]: I wonder if we shouldn't make it v1. I wonder if it makes any difference at all right now.

File: spot-electron/src/config/config.js [hunk 10]
```diff
@@ -0,0 +1,92 @@
+const { app } = require('electron');
+const fs = require('fs');
+const _ = require('lodash');
+const { logger } = require('../logger');
+const defaultConfig = require('../../config');
+
+const PERSISTED_CONFIG = `${app.getPath('userData')}/config.json`;
```
Comment (virtuacoplenny) [inline spot-electron/src/config/config.js]: Do you if this type of path accessing with forward slash also work on windows? (not saying change it now if needed)

File: spot-electron/.eslintrc (hunk 7 above)
Comment (zbettenbuk) [reply spot-electron/.eslintrc]: I reused your file :D

File: spot-electron/config.js (hunk 8 above)
Comment (zbettenbuk) [reply spot-electron/config.js]: Can be, not now. Yana said we should use this for now

Comment (saghul) [reply spot-electron/config.js]: Me no likey. It doesn't make sense that this is not the Jitsi logo. There is no precedent, even.

Comment (zbettenbuk) [reply spot-electron/config.js]: Removed, as well as replaced the default URL

File: spot-electron/src/config/config.js (hunk 10 above)
Comment (zbettenbuk) [reply spot-electron/src/config/config.js]: Good catch, I should have used node path

File: spot-electron/package.json (hunk 9 above)
Comment (zbettenbuk) [reply spot-electron/package.json]: I think this is something yet to sort out. I'm okay with any kind of versioning

Comment (virtuacoplenny) [reply spot-electron/package.json]: i just remembered we do date versioning (YYYY.MM.DD) for spot web releases. We should be doing that for mobile as well so probably we'll end up doing that for electron if possible.

**Pull request merged**